handler_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connect
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/bufbuild/connect-go/internal/assert"
+	pingv1 "github.com/bufbuild/connect-go/internal/gen/connect/ping/v1"
+)
+
+func TestHandler_ServeHTTP(t *testing.T) {
+	t.Parallel()
+	const procedure = "/connect.ping.v1.PingService/Ping"
+	handler := NewUnaryHandler[pingv1.PingRequest, pingv1.PingResponse](
+		procedure,
+		func(ctx context.Context, req *Request[pingv1.PingRequest]) (*Response[pingv1.PingResponse], error) {
+			return nil, NewError(CodeUnimplemented, nil)
+		})
```

Comment:
If you move this out to `connect_test`, I think you can use the generated code here - create a mux, use `NewPingServiceHandler` with `UnimplementedPingServiceHandler`, etc. We'd also be able to use the generated constant for the fully-qualified service name (the first part of `procedure`).